doc/concepts/pipeline-concepts/datum/relationship-between-datums.md

```diff
@@ -0,0 +1,56 @@
+
+# Relationship Between Datums
```

Comment:
I can see what you're going for in this section -- cribbing from the opening part of "lifecycle of a datum", but unfortunately almost all the information here is incorrect and does not teach the concept we're going for. "Incremental processing" and "relationship between input and output datums" are fundamentally linked. I think we should actually call this whole doc Incremental processing and redesign how it fits together with lifecycle of a datum.

I completely agree that the Lifecycle of a Datum doc is too complicated, too long, and doesn't explain the concepts cleanly. We should do a TOH to break down that doc correctly and decide what should actually go into this "relationship between input and output datums" docs. Unfortunately, this means that we need to start from scratch on this doc.